### PR TITLE
[MoE] Fix llama4 calibration, bump transformers

### PR DESCRIPTION
**Pull request by kylesayrs**
## Purpose ##
It was noticed that llama4 calibration tests failed after weights were initialized with a normal distribution. This revealed that outputs from the calibrated module were slightly different than those of the original model.

The issue was that llama4 applies router scores **before** experts, whereas most MoEs apply router scores after experts.

## Changes ##
* Fix llama4 model definition to scale by router scores before experts are applied
* Drop support for `transformers<4.53` (current version is 4.57)

## Testing ##
* Updated llama4 test to use normally initialized weights and to check router logits as well as normal outputs.

### CODE REVIEW

File: src/llmcompressor/modeling/llama4.py [hunk 1]
```diff
@@ -33,39 +33,28 @@ def __init__(
...
-                router_logits, self.top_k, dim=1
-            )
-            router_logits = router_outputs
-            router_scores = (
-                torch.full_like(router_logits, float("-inf"))
-                .scatter_(1, router_indices, router_top_value)
-                .transpose(0, 1)
-            )
-            router_scores = torch.sigmoid(router_scores.float()).to(hidden_states.dtype)
+        router_scores, router_logits = self.router(hidden_states)  # transformers>=4.53
```
Comment (dsikka) [inline src/llmcompressor/modeling/llama4.py]: We either have to bump our transformers version support in setup.py or make sure we're supporting < 4.53.

File: tests/llmcompressor/modeling/test_calib_llama4.py [hunk 2]
```diff
@@ -68,21 +68,25 @@ def test_calib_llama4_module():
...
+        assert torch.nn.functional.mse_loss(true_out, out) < 1e-10
+        assert torch.nn.functional.mse_loss(true_router_logits, router_logits) < 1e-10
 
     module = SequentialLlama4TextMoe(config, original, calibrate_all_experts=False)
     with calibration_forward_context(module):
-        output = module(sample)[0]
-        assert torch.allclose(true_output, output, atol=1e-6)
+        out, router_logits = module(sample)
+        assert torch.nn.functional.mse_loss(true_out, out) < 1e-10
+        assert torch.nn.functional.mse_loss(true_router_logits, router_logits) < 1e-10
```
[the remark this answers is not in the document]
Comment (rahul-tuli) [reply tests/llmcompressor/modeling/test_calib_llama4.py:92]: nit: Makes sense?

File: src/llmcompressor/modeling/llama4.py (hunk 1 above)
Comment (rahul-tuli) [reply src/llmcompressor/modeling/llama4.py]: Why <4.53? as of now we support `transformers>=4.53.0,<=4.56.2` we should try to bump up

Comment (kylesayrs) [reply src/llmcompressor/modeling/llama4.py]: Why would we need to increase the support version when 4.53.0 is already the minimum.

Comment (dsikka) [reply src/llmcompressor/modeling/llama4.py]: Sorry, initial comment was unclear 
Iirc, our lower bound transformers version uses a slightly different llama4 definition. It returns router_scores, not logits. 

I am ok with bumping up the lower bound but we will need to do that in setup.py if we’re not longer supporting the older definition

Comment (kylesayrs) [reply src/llmcompressor/modeling/llama4.py]: My mistake, the original comment was wrong, we require `transformers>=4.54`

Comment (dsikka) [reply src/llmcompressor/modeling/llama4.py]: yeah sorry, I should have said that initially.
